### PR TITLE
Sla enhance view

### PR DESCRIPTION
**Pull request by orthagh**
Enhancement of #2185 

Contains:

- a more simple (and condensed) ticket form: 
image

- a timeline into stats tab which includes new slm dates:
image

Possible TODO:

- [x] transform sla into links in the timeline
- [x] add a new component in Html for reusable datetimelines (with a tests)
- [ ] remove some dates (mostly internals) to main ticket form and set them into stats tab (renamed)
- [ ] permit to add step dates (sla ones) into this new tab (remove completly sla management from main form)
- [x] big one, we need a big refactor into sla/ola class

### CODE REVIEW

File: inc/html.class.php [hunk 1]
```diff
@@ -3596,6 +3596,74 @@ static function computeGenericDateTimeSearch($val, $force_day = false, $specific
...
+            $out.= "<time>".date("Y-m-d H:i:s", $data['timestamp'])."</time>";
+            $out.= "<span class='dot'></span>";
+            $out.= "<label>".$data['label']."</label>";
+            $out.= "</li>";
+         }
+      }
+      $out.= "</ul>";
+      $out.= "</div>";
+
+      if ($options['output']) {
```
Comment (trasher) [inline inc/html.class.php]: Generally, "display" is used rather than "output".

File: inc/levelagreement.class.php [hunk 2]
```diff
@@ -0,0 +1,1056 @@
...
+    * @param integer $calendars_id calendars_id of the ticket
+   **/
+   function setTicketCalendar($calendars_id) {
+
+      if ($this->fields['calendars_id'] == -1) {
+         $this->fields['calendars_id'] = $calendars_id;
+      }
+   }
+
+   function getFromDB($ID) {
```
Comment (trasher) [inline inc/levelagreement.class.php]: Use `post_getFromDB`?

File: inc/levelagreement.class.php [hunk 3]
```diff
@@ -0,0 +1,1056 @@
...
+            Html::showMassiveActions($massiveactionparams);
+            Html::closeForm();
+         }
+      } else {
+         echo __('No item to display');
+      }
+      echo "</div>";
+   }
+
+   function showRulesList() {
```
Comment (trasher) [inline inc/levelagreement.class.php]: missing phpdoc

**Comment by trasher** [comment]
I tested quickly; seems it is OK.

**Comment by orthagh** [comment]
review comments fixed

**Pull request merged**